GliaWidgets/Sources/CallVisualizer/VideoCall/View/ConnectView/VideoCallView.ConnectView.swift

```diff
@@ -0,0 +1,149 @@
+                left: 0,
+                bottom: 10,
+                right: 0
+            )
+        }
+
+        var props: Props = Props() {
+            didSet {
+                renderProps()
+            }
+        }
+
+        var renderShowing: CallVisualizer.VideoCallViewModel.Transaition = .hide(animation: false) {
+            didSet {
+                guard renderShowing != oldValue else { return }
+                switch renderShowing {
+                case let .show(animation):
+                    show(animated: animation)
+                case let .hide(animation):
+                    hide(animated: animation)
+                }
+            }
+        }
+
+        private lazy var stackView = UIStackView().make {
+            $0.axis = .vertical
+            $0.spacing = 1
+            $0.distribution = .fillProportionally
+        }
+
+        // MARK: - Overrides
+
+        override func setup() {
+            super.setup()
+            accessibilityElements = [operatorView, statusView]
+            addSubview(stackView)
+            stackView.addArrangedSubviews([operatorView, statusView])
+            stackView.translatesAutoresizingMaskIntoConstraints = false
+        }
+
+        override func defineLayout() {
+            super.defineLayout()
+            contentTopPadding = stackView.autoPinEdgesToSuperviewEdges(with: CallVisualizer.VideoCallView.ConnectView.contentInsets).first
+        }
+    }
+}
+
+// MARK: - Props
+extension CallVisualizer.VideoCallView.ConnectView {
+    struct Props: Equatable {
+        let operatorViewProps: CallVisualizer.VideoCallView.ConnectOperatorView.Props
+        let statusViewProps: CallVisualizer.VideoCallView.ConnectStatusView.Props
+        let state: CallVisualizer.VideoCallViewModel.State
+        let transition: CallVisualizer.VideoCallViewModel.Transaition
+
+        init(
+            operatorViewProps: CallVisualizer.VideoCallView.ConnectOperatorView.Props = .init(),
+            statusViewProps: CallVisualizer.VideoCallView.ConnectStatusView.Props = .init(),
+            state: CallVisualizer.VideoCallViewModel.State = .initial,
+            transition: CallVisualizer.VideoCallViewModel.Transaition = .hide(animation: false)
+        ) {
+            self.operatorViewProps = operatorViewProps
+            self.statusViewProps = statusViewProps
+            self.state = state
+            self.transition = transition
+        }
+    }
+}
+
+extension CallVisualizer.VideoCallView.ConnectView.Props {
+
+    enum Layout {
+        case chat, call
+    }
+}
+
+// MARK: - Private
+
+private extension CallVisualizer.VideoCallView.ConnectView {
+    func renderProps() {
+        operatorView.props = props.operatorViewProps
+        statusView.props = props.statusViewProps
+        contentSpacing(for: props.state)
+        topPadding(for: props.state)
+        renderShowing = props.transition
+        updateConstraints()
+
+    }
+
+    func show(animated: Bool) {
```

Comment:
To not avoid code duplication you can add intermediate method looks like
```swift 
func show(animated: Bool) {
        animate(animated: animated) {
            self.transform = .identity
        }
    }

    func hide(animated: Bool) {
        animate(animated: animated) {
            self.transform = CGAffineTransform(scaleX: 0, y: 0)
        }
    }

    func animate(
        animated: Bool,
        animations: @escaping () -> Void
    ) {
        UIView.animate(
            withDuration: animated ? 0.5 : 0.0,
            delay: 0.0,
            usingSpringWithDamping: 0.8,
            initialSpringVelocity: 0.7,
            options: .curveEaseInOut,
            animations: animations,
            completion: nil
        )
    }
```

Reply:
fixed [00e2997](https://github.com/salemove/ios-sdk-widgets/pull/494/commits/00e29971b62e4284d2a0668fd9828bc9cb1e0b78)